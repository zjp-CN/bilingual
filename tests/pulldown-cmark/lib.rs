#[cfg(test)]
mod tests {
    use insta::{assert_display_snapshot, assert_yaml_snapshot};
    use pulldown_cmark::{CowStr, Event, Options, Parser, Tag};

    pub fn filter(event: &Event, include: &mut bool) -> bool {
        match event {
            Event::Start(Tag::CodeBlock(_)) => {
                *include = false;
                false
            }
            Event::End(Tag::CodeBlock(_)) => {
                *include = true;
                false
            }
            Event::Start(Tag::Heading(_)) | Event::Start(Tag::Paragraph) => false,
            // 排除行间代码
            _ => *include,
        }
    }

    pub fn extract<'a>(event: Event<'a>) -> CowStr<'a> {
        match event {
            Event::Text(x) => x,
            Event::End(Tag::Heading(_)) | Event::End(Tag::Paragraph) => "\n".into(),
            Event::Code(x) => format!("\n{}\n", x).into(), // 行内代码

            _ => " ".into(),
        }
    }

    #[test]
    fn default_config() -> std::io::Result<()> {
        let file = "8_6_io_eventqueue";
        let md = std::fs::read_to_string(format!("../../assets/{}.md", file))?;
        assert_yaml_snapshot!(file, Parser::new(&md).collect::<Vec<_>>());
        assert_yaml_snapshot!(format!("{}_offset", file),
                              Parser::new(&md).into_offset_iter().collect::<Vec<_>>());

        let parsed = {
            let mut include = true;
            Parser::new(&md).filter(|event| filter(event, &mut include))
                            .map(extract)
                            .collect::<Vec<_>>()
        };
        // assert_yaml_snapshot!("8_6_io_eventqueue_modified",
        // Parser::new(&md).collect::<Vec<_>>());
        assert_yaml_snapshot!(format!("{}_modified", file), parsed.join(""));

        assert_yaml_snapshot!("8_6_io_eventqueue_translated",
                              Parser::new(&std::fs::read_to_string("../../assets/\
                                                                    8_6_io_eventqueue_translated.\
                                                                    md")?).collect::<Vec<_>>());

        Ok(())
    }

    const OPT: Options = Options::all();

    #[test]
    fn full_config() -> std::io::Result<()> {
        let file = "markdown-it";
        let md = std::fs::read_to_string(format!("../../assets/{}.md", file))?;
        assert_yaml_snapshot!(file, Parser::new_ext(&md, OPT).collect::<Vec<_>>());
        assert_yaml_snapshot!(format!("{}_offset", file),
                              Parser::new_ext(&md, OPT).into_offset_iter().collect::<Vec<_>>());

        let parsed = {
            let mut include = true;
            Parser::new_ext(&md, OPT).filter(|event| filter(event, &mut include))
                                     .map(extract)
                                     .collect::<Vec<_>>()
        };
        // assert_yaml_snapshot!("8_6_io_eventqueue_modified",
        // Parser::new(&md).collect::<Vec<_>>());
        assert_display_snapshot!(format!("{}_modified", file), parsed.join(""));

        Ok(())
    }
}
