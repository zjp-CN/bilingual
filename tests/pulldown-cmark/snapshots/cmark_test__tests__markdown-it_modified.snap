---
source: tests/pulldown-cmark/lib.rs
expression: "parsed.join(\"\")"

---
  Advertisement :) 
    pica   - high quality and fast image resize in browser.    babelfish   - developer friendly i18n with plurals support and easy syntax.  You will like those projects!
 h1 Heading 8-)
h2 Heading
h3 Heading
h4 Heading
h5 Heading
h6 Heading
Horizontal Rules
   quotation marks
"Smartypants, double quotes" and 'single quotes'
Emphasis
 This is bold text 
 This is bold text 
 This is italic text 
 This is italic text 
 Strikethrough 
Blockquotes
 Blockquotes can also be nested...
 ...by using additional greater-than signs right next to each other...
 ...or with spaces between arrows.
   Lists
Unordered
  Create a list by starting a line with 
+
, 
-
, or 
*
  Sub-lists are made by indenting 2 spaces:  Marker character change forces new list start:  Ac tristique libero volutpat at    Facilisis in pretium nisl aliquet    Nulla volutpat aliquam velit      Very easy!  Ordered
  Lorem ipsum dolor sit amet
  Consectetur adipiscing elit
  Integer molestie lorem at massa
  You can use sequential numbers...
  ...or keep all the numbers as 
1.

  Start numbering with offset:
  foo  bar  Code
Inline 
code

Indented code
Block code "fences"
Syntax highlighting
Tables
   Option  Description    data  path to data files to supply the data that will be passed into templates.    engine  engine to be used for processing templates. Handlebars is the default.    ext  extension to be used for dest files.   Right aligned columns
   Option  Description    data  path to data files to supply the data that will be passed into templates.    engine  engine to be used for processing templates. Handlebars is the default.    ext  extension to be used for dest files.   Links
 link text 
 link with title 
 ref link 
Autoconverted link https://github.com/nodeca/pica (enable linkify to see)
Images
 Minion   Stormtroopocat 
Like links, Images also have a footnote style syntax
 Alt text 
With a reference later in the document defining the URL location:
Plugins
The killer feature of 
markdown-it
 is very effective support of  syntax plugins .
 Emojies 
 Classic markup: :wink: :crush: :cry: :tear: :laughing: :yum:
Shortcuts (emoticons): :-) :-( 8-) ;)
 see  how to change output  with twemoji.
 Subscript  /  Superscript 
  19^th^  H~2~O   <ins> 
++Inserted text++
 <mark> 
==Marked text==
 Footnotes 
Footnote 1 link .
Footnote 2 link .
Inline footnote^[Text of inline footnote] definition.
Duplicated footnote reference .
 Footnote  can have markup 
  Footnote text.
  Definition lists 
Term 1
:   Definition 1 with lazy continuation.
Term 2 with  inline markup 
:   Definition 2
 Compact style: 
Term 1 ~ Definition 1
Term 2 ~ Definition 2a ~ Definition 2b
 Abbreviations 
This is HTML abbreviation example.
It converts "HTML", but keep intact partial entries like "xxxHTMLyyy" and so on.
*[HTML]: Hyper Text Markup Language
 Custom containers 
::: warning  here be dragons  :::

